packages/eslint-plugin/src/rules/no-unsafe-function-type.ts

```diff
@@ -0,0 +1,46 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+
+import { createRule } from '../util';
+
+export default createRule({
+  name: 'no-unsafe-function-type',
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow using the unsafe built-in Function type',
+      recommended: 'recommended',
+    },
+    fixable: 'code',
+    messages: {
+      bannedFunctionType: [
+        'The `Function` type accepts any function-like value.',
+        'Prefer explicitly defining any function parameters and return type.',
+      ].join('\n'),
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    function checkBannedTypes(node: TSESTree.Node): void {
+      if (node.type === AST_NODE_TYPES.Identifier && node.name === 'Function') {
+        context.report({
+          node,
+          messageId: 'bannedFunctionType',
+        });
+      }
+    }
+
+    return {
+      TSClassImplements(node): void {
```

Comment:
Thought: should the two new rules check `implements` and `extends`? You can't extend `number` or `() => {}` so the use of `Number` must either be intentional or refer to another type (since we still don't have scope analysis for this rule)

Reply:
Someone could, in theory, be doing:
* `extends`: `interface MyFunction extends Function { __brand: 'my-function'; }`
* `implements`: `class Weird implements Function {}`. 

...so for completeness, I think we need to?

Reply:
Is that a bad idea? If they do it there's no reason they are doing it by accident or ignorance.

Reply:
I'm not strongly opposed to removing this for simplicity. It does feel a little weird to me to leave in a known thing that nobody should really ever do - even if it's only doable with great intent.

Reply:
Let's leave them in for now, but at least `extends Function` feels fine to me.